src/components/Badge/BadgeReport.jsx

```diff
@@ -217,20 +217,28 @@ const BadgeReport = (props) => {
 
   const deleteBadge = () => {
     let newBadges = sortBadges.slice();
-    newBadges.splice(badgeToDelete, 1);
+    const deletedBadge = newBadges.splice(badgeToDelete, 1);  
+    if (deletedBadge[0].featured) {
+      setNumFeatured(--numFeatured);
+    } 
     setSortBadges(newBadges);
     setShowModal(false);
     setBadgeToDelete(null);
   };
 
   const saveChanges = async () => {
-    let newBadgeCollection = sortBadges.slice();
+    let newBadgeCollection = JSON.parse(JSON.stringify(sortBadges));  
     for (let i = 0; i < newBadgeCollection.length; i++) {
       newBadgeCollection[i].badge = newBadgeCollection[i].badge._id;
-    }
+    }  
     console.log(newBadgeCollection);
     await props.changeBadgesByUserID(props.userId, newBadgeCollection);
     await props.getUserProfile(props.userId);
+    
+    props.setUserProfile(prevProfile => {
+      return {...prevProfile, badgeCollection: sortBadges }
+    });
+    props.handleSubmit();
```

Comment:
This is a function passed from UserProfile.jsx that is used for when 'saving' the user profile. Using setUserProfile wasn't enough alone to stop old stale data from coming back after manually refreshing the browser. I know we're trying to move a lot of functions from UserProfile to get it from being a 'monolith', but I doubt this important centralized function will be moved. Let me know if this is not the case.